app/src/main/java/org/mozilla/tv/firefox/settings/SettingsFragment.kt

```diff
@@ -93,6 +105,52 @@ class SettingsFragment : Fragment() {
         return view
     }
 
+    private fun setupFxaProfileScreen(
+        inflater: LayoutInflater,
+        parentView: ViewGroup?
+    ): View {
+        val view = inflater.inflate(R.layout.settings_screen_fxa_profile, parentView, false)
+
+        setupFxaProfileListeners(view)
+        profileDisposable = observeFxaProfile(view)
+
+        return view
+    }
+
+    private fun setupFxaProfileListeners(view: View) {
+        view.button_firefox_tabs.setOnClickListener {
+            // TODO perform action
+            // TODO telemetry
+        }
+        view.button_sign_out.setOnClickListener {
+            // TODO perform action
+            // TODO telemetry
+            serviceLocator!!.screenController.handleBack(fragmentManager!!)
+        }
+        view.back_button.setOnClickListener {
+            // TODO telemetry
+            serviceLocator!!.screenController.handleBack(fragmentManager!!)
+        }
+    }
+
+    private fun observeFxaProfile(view: View): Disposable = context!!.serviceLocator.fxaRepo.accountState
+        .ofType(FxaRepo.AccountState.AuthenticatedWithProfile::class.java)
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe {
+            view.description_bottom.text = it.profile.displayName
```

Comment:
This is probably ridiculous, but my thinking was that the layout could well be reused in the future with top text / bottom text, whereas display name is coupled to this use case.  But I'm fine changing it, it's unlikely to happen.

Reply:
These days I lean towards, "Do the simple, easy-to-change thing, specific to the current use case, because it's hard to predict when generalizing will be useful and it creates indirection." But I understand the line is blurry sometimes. :)